src/Microsoft.TemplateEngine.Cli/Commands/create/InstantiateCommand.Help.cs

```diff
@@ -431,26 +435,67 @@ private static IEnumerable<TemplateOption> CollectOptionsToShow(IEnumerable<Temp
             return optionsToShow;
         }
 
-        private void WriteCustomInstantiateHelp(HelpContext context)
+        private static void WriteCustomInstantiateHelp(HelpContext context, Command command)
         {
+            //unhide arguments of NewCommand. They are hidden not to appear in subcommands help.
+            foreach (Argument arg in command.Arguments)
```

Comment:
Can you give some more context? Why are they appearing on subcommands?

Reply:
This is a bit of the hack. 
By default, the usage will list arguments for all commands.
So, if we keep arguments on new level, the usage for uninstall as example will be:
`Usage:`
  `dotnet-new3 new3 [<template-short-name> [<template-args>...]] uninstall [<package>...] [options]`

and it is not valid, so I unhide it only in case showing help for `new`. 
for instantiate command (`create`) it will do nothing, as arguments are visible there.

Reply:
Is this a bug in library? If so, can you reference issue in comment?

Reply:
we discussed it some time ago, it's not a bug. For some commands it is a valid syntax. 

For example if we ever could install the template by short name
`dotnet new my-template install` or `dotnet new my-template uninstall` might be a reasonable syntax. 

It's rather a missing feature not to show parent command arguments when not needed.